File: food-review/src/main/java/com/food/review/usecase/exception/ValidationException.java
package com.food.review.usecase.exception;

import com.food.review.domain.error.FieldError;
import lombok.Getter;

import java.util.ArrayList;
import java.util.List;

public class ValidationException extends BaseException {

    private static final long serialVersionUID = 6195373487710917070L;

    @Getter
    private final transient List<FieldError> fieldErrors = new ArrayList<>();

    public ValidationException(String code, String message) {
        super(code, message);
    }

    public ValidationException(String code, String message, List<FieldError> constraintsErrors) {
        super(code, message);
        this.fieldErrors.addAll(constraintsErrors);
    }
}

File: food-restaurant/src/main/kotlin/com/food/restaurant/usecase/exception/ValidationException.kt
package com.food.restaurant.usecase.exception

class ValidationException(
        val code: String,
        val error: String,
        private val description: String,
        val messageArgs: List<String>
): RuntimeException(description) {

    constructor(code: String, error: String, description: String) :
        this(code, error, description, listOf())

}
